poglink/cogs/rates.py

```diff
@@ -113,73 +118,70 @@ async def send_embed(self, description, url):
+                    logger.info(
+                        f"Rates for {url} changed. Resetting consecutive rates count to 1"
+                    )
+                else:
+                    self.consecutive_count[idx] += 1
+                    logger.info(
+                        f"Rates for {url} unchanged. Consecutive count = {self.consecutive_count[idx]}"
+                    )
 
-        return rates_diff
+                if self.consecutive_count[idx] == 2:
+                    logger.info(
+                        f"New rates have become stable. Comparing against previous stable rates."
+                    )
+                    if self.stable_rates[idx]:
+                        stable_diff = self.stable_rates[idx].get_diff(current_rates)
+                        if stable_diff.items:
+                            # generate and send embed
+                            logger.info(
+                                f"Rates at {url} changed since last stable value - sending embed"
+                            )
+                            embed_description = stable_diff.to_embed()
+                            await self.send_embed(embed_description, url)
+                    else:
+                        logger.info(
+                            "No previous stable rates recorded. Updating new stable value, but no updates to publish."
+                        )
+
+                    self.stable_rates[idx] = current_rates
+            else:
+                logger.info(f"No previous rates stored yet; skipping.")
+
+            # Update last rates value for next iteration
+            self.last_rates[idx] = current_rates
 
     # Events
     @commands.Cog.listener()
     async def on_ready(self):
         logger.info("Cog Ready: Rates")
 
         while True:
-            for url, output_path in zip(self.webpage_urls, self.output_paths):
-                try:
-                    rates_diff = await self.compare_posted_rates(url, output_path)
-                except RatesFetchError as e:
-                    logger.error(
-                        f"Could not retrieve rates from ARK Web API at {url}: {e}"
-                    )
-                except RatesProcessError as e:
-                    logger.error(f"Could not process rates URL {url}: {e}")
-                except RatesWriteError as e:
-                    logger.error(f"Failed to write rates to {output_path}: {e}")
-                except Exception as e:
-                    logger.error(e)
-
-                if rates_diff.items:
-                    # generate and send embed
-                    logger.info(f"Rates at {url} changed - sending embed")
-                    embed_description = rates_diff.to_embed()
-                    await self.send_embed(embed_description, url)
-                else:
-                    logger.debug(f"No change in rates at {url}.")
-
-                await asyncio.sleep(self.polling_delay)
+            await self.compare_and_notify_all()
```

Comment:
Note, I just broke the main logic out into a separate function here so it can be tested on its own in tests. This way we can avoid having to run an instance of the bot and waiting for it to go through the polling delay (kinda like a "spin once" instead of "spin").

That's all that I've changed here since the last time you looked at this PR, but this change just cluttered up the diff a little bit where it's substituting the old `compare_posted_rates()` with this new `compare_and_notify_all()`.